crates/bevy_ecs/src/world/entity_fetch.rs

```diff
@@ -3,12 +3,98 @@ use core::mem::MaybeUninit;
 
 use crate::{
     entity::{hash_map::EntityHashMap, hash_set::EntityHashSet, Entity, EntityDoesNotExistError},
+    error::Result,
     world::{
         error::EntityMutableFetchError, unsafe_world_cell::UnsafeWorldCell, EntityMut, EntityRef,
         EntityWorldMut,
     },
 };
 
+/// Provides a safe interface for non-structural access to the entities in a [`World`].
+///
+/// This cannot add or remove components, or spawn or despawn entities,
+/// making it relatively safe to access in concert with other ECS data.
+/// This type can be constructed via [`World::entities_and_commands`],
+/// or [`DeferredWorld::entities_and_commands`].
+///
+/// [`World`]: crate::world::World
+/// [`World::entities_and_commands`]: crate::world::World::entities_and_commands
+/// [`DeferredWorld::entities_and_commands`]: crate::world::DeferredWorld::entities_and_commands
+pub struct EntityFetcher<'w> {
```

Comment:
I could be misremembering, but don't we already have a ton of entity like structs in our codebase? Not saying we don't need this one, but just something to consider.

Reply:
This isn't an entity-like struct, it's a struct for fetching entity-likes. Not sure what you mean.